source/cppfront.cpp

```diff
@@ -346,6 +353,8 @@ class positional_printer
             //  remain diff-identical
             print_extra("\n");
         }
+
+        cpp1_filename.empty() ? std::cout<<out.str() : file_out<<out.str();
```

Comment:
I would use a regular if/else statement here. None of the current code use the ternary operator.